### DISCUSSION TITLE
Reinitialize `useForm` instance

### DISCUSSION THREAD

**Discussion by noahsb** [mantinedev/mantine#3290] [Q&A] [answered]
Is there a way to "reinitialize" `useForm` to reset the form completely? 

`form.reset()` gets close, but I'd like to reinitialize the form so that the `initialValues` get re-cached. The reason for this is because the form submission I send out is different from what gets returned by the api. 

`form.setValues` is a little tedious to use for this particular case because I don't have direct access to the data. I could solve this by adding a `useEffect`, but curious if there's an alternative before I go down that route

Any suggestions on how I might accomplish this?

**Comment by rtivital** [comment, marked as answer]
You can unmount/mount component that calls useForm hook

**Reply by noahsb** [reply]
Good point, thanks!
